Broken build: missing openssl include?

At commit 3785fab1625ead21d8481ade60e530d7eba3ec51

```
$ make -j
g++ --std=c++11 -fno-strict-aliasing -Wall -c -I. -I./include -I/usr/include/ -I./src/ -g -ggdb src/hashutil.cc -o src/hashutil.o
src/hashutil.cc: In function ‘EVP_MD_CTX* cuckoofilter::EVP_MD_CTX_new()’:
src/hashutil.cc:719:44: error: ‘OPENSSL_zalloc’ was not declared in this scope
    return OPENSSL_zalloc(sizeof(EVP_MD_CTX));
                                            ^
Makefile:27: recipe for target 'src/hashutil.o' failed
make: *** [src/hashutil.o] Error 1

```

@florianjacob

```
$ make
g++ --std=c++11 -fno-strict-aliasing -Wall -c -I. -I./include -I/usr/include/ -I./src/ -g -ggdb src/hashutil.cc -o src/hashutil.o
src/hashutil.cc: In function ‘void* cuckoofilter::OPENSSL_zalloc(size_t)’:
src/hashutil.cc:722:25: error: ‘memset’ was not declared in this scope
       memset(ret, 0, num);
                         ^
src/hashutil.cc: In function ‘EVP_MD_CTX* cuckoofilter::EVP_MD_CTX_new()’:
src/hashutil.cc:728:25: error: invalid conversion from ‘void*’ to ‘EVP_MD_CTX* {aka env_md_ctx_st*}’ [-fpermissive]
    return OPENSSL_zalloc(sizeof(EVP_MD_CTX));
                         ^
Makefile:27: recipe for target 'src/hashutil.o' failed
make: *** [src/hashutil.o] Error 1
```

--- a/src/hashutil.cc
+++ b/src/hashutil.cc
@@ -726,7 +726,7 @@ static void *OPENSSL_zalloc(size_t num)
 
 EVP_MD_CTX *EVP_MD_CTX_new(void)
 {
-   return OPENSSL_zalloc(sizeof(EVP_MD_CTX));
+   return (EVP_MD_CTX *)OPENSSL_zalloc(sizeof(EVP_MD_CTX));
 }
 
 void EVP_MD_CTX_free(EVP_MD_CTX *ctx)

--- a/src/hashutil.cc
+++ b/src/hashutil.cc
@@ -714,6 +714,7 @@ uint32_t HashUtil::NullHash(const void *buf, size_t length,
  * Implemented as proposed by https://wiki.openssl.org/index.php/OpenSSL_1.1.0_Changes
  */
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
+#include <string.h>
 static void *OPENSSL_zalloc(size_t num)
 {
   void *ret = OPENSSL_malloc(num);

--- a/src/hashutil.cc
+++ b/src/hashutil.cc
@@ -714,6 +714,15 @@ uint32_t HashUtil::NullHash(const void *buf, size_t length,
  * Implemented as proposed by https://wiki.openssl.org/index.php/OpenSSL_1.1.0_Changes
  */
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
+static void *OPENSSL_zalloc(size_t num)
+{
+  void *ret = OPENSSL_malloc(num);
+
+  if (ret != NULL)
+      memset(ret, 0, num);
+  return ret;
+}
+
 EVP_MD_CTX *EVP_MD_CTX_new(void)
 {
    return OPENSSL_zalloc(sizeof(EVP_MD_CTX));

